### PR TITLE
Release/0.4.0

### LINKED ISSUE #21: Flexible config
It would be nice if you could pass arbitrary configs to the config file. You may want to publish to S3 or do something else that requires config beyond the basics the cookbook currently has. Perhaps the config should be driven by a resource that could take in a config hash.

### PR DESCRIPTION
**Pull request by YpNo**
Release: 0.4.0

- Big refacto
- Update resource with latest usage of aptly
- Create aptly_serve and aptly_api_serve resource
- Create unit tests
- Enhance integration test
- Use Circle CI

I think this release fix #21 #25 and #32 issues.

### CODE REVIEW

**Comment by damacus** [comment]
I'm not sure what's happened here, it's throw away a load of circleci config?

File: .kitchen.vagrant.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
```
Comment (damacus) [inline .kitchen.vagrant.yml]: you shouldn't need this? `kitchen.yml` should provide this.

File: .kitchen.yml [hunk 2]
```diff
@@ -1,22 +1,44 @@
+---
 driver:
-  name: vagrant
+  name: dokken
```
Comment (damacus) [inline .kitchen.yml]: the dokken config is typically kept in the `kitchen.dokken.yml` files (checkout haproxy or postgresql for a more up to date config)

File: metadata.rb [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
+description 'Installs/Configures aptly'
+long_description 'Installs/Configures aptly'
+version '0.4.0'
+chef_version '>= 12.14' if respond_to?(:chef_version)
 
 supports 'ubuntu'
 supports 'debian'
 
-depends 'apt'
+# depends 'ark'
```
Comment (damacus) [inline metadata.rb]: This can just get removed.

File: .kitchen.vagrant.yml (hunk 1 above)
Comment (YpNo) [reply .kitchen.vagrant.yml]: Maybe not :)
You can aggregate dokken and vagrant config as the same time ?

File: metadata.rb (hunk 3 above)
Comment (YpNo) [reply metadata.rb]: it's for a future improve (tarball installation). I can remove it from PR

File: .kitchen.yml (hunk 2 above)
Comment (YpNo) [reply .kitchen.yml]: Ok but how can I choose the default driver for my local and circle build ?

**Pull request closed without merging**